### DISCUSSION TITLE
Butler not showing all the informations About the Task Scheduled Event

### DISCUSSION THREAD

**Discussion by wieeem** [ptarmiganlabs/butler#728] [Q&A] [answered]
Hello All, 

I am using butler to get task scheduler logs from my qliksense server.

I configured Butler with LogEvent : 
![logevent](https://github.com/ptarmiganlabs/butler-sos/assets/47701160/46e76ba9-2c5b-4d95-b0df-75948cf7b7cd)

I added also the LocalLogConfig.xml under C:\ProgramData\Qlik\Sense\Scheduler 

The problem is that Butler is retrieving incomplete info about the Task: 
![MicrosoftTeams-image (3)](https://github.com/ptarmiganlabs/butler-sos/assets/47701160/ff044258-a1d5-4eab-b8ce-e07cc3c6b83f)

(task_name:"",app_name:"",exception_message:"",........)

Anyone has an idea please ?

**Comment by mountaindude** [comment, marked as answer]
Hi, thanks for reaching out.

First, you mention "Butler" and that you are interested in scheduler error log entries, which leads me to suspect you are interested in failed reloads..?  If that's the case the Butler tool is a better choice than Butler SOS.

The "Butler" tool (butler.ptarmiganlabs.com) deals with things such as failed reload alerts, scheduling, key-value store and much more, whereas "Butler SOS" (butler-sos.ptarmiganlabs.com) deals with real-time monitoring of Qlik Sense - both health/load metrics and real-time delivery of select log entries. 

So... If you have Butler SOS running and the scheduler service writes an error row to it's log file, that _should_ be forwarded to Butler SOS. 
And it seems that's happening as you do get a message in Butler SOS, just that the "exception_message" is empty.
But the fact that you do get a message in Butler SOS means the XML log appender file is working: It is sending a message to Butler SOS.

Ok, next step is then to figure out why the exception_message is empty.

Might sound like an easy thing, but this is in fact a big can of worms... There is no documentation what so ever from Qlik about this.
I always end up comparing the actual warning/error row in the log file with what I see in Butler SOS or Butler (both use the same XML appender concept).
If the exception_message is empty in the log file row (which I suspect is the case here) you did in fact get the correct data sent to Butler SOS: an empty string.

If there on the other hand **IS** an exception_message in the Scheduler's log file, for the warning/error line in question, then something is broken. Either you need to tweak the XML file to trigger on some other text pattern in the log file, or something else is causing this.

Bottom line: Find the correct row in the log file and compare it with what you see in Butler SOS.
If you can share that log line (you may want to mask sensitive info if there is any) I'd be happy to take a look. 
We can then decide what the next step is.

**Comment by wieeem** [comment]
Hi, Thank you for your reply.

I switched to Butler instead of butler-sos as you suggested and it works perfectly. 

I used teamsNotifications to get the scheduled failed task logs. and I would actually like to retrieve the app owner email with the logs.

I saw that it's possible to retrieve the AppOwner with Emailnotification, but unfortunately I can't use this feature at the moment.

So I wanted to know if it will be possible to inject the Appowner into the body of the Teams template.

Thanks in advance

**Comment by mountaindude** [comment]
Good that you got Butler working.

App owner is not currently available as a template field.  
The available fields are listed here: https://butler.ptarmiganlabs.com/docs/reference/alert-template-fields/

But you are right - app owner email _is_ available for emails, or at least it's possible to (optionally) send alert emails to app owners. 
So the data is there, but it's not exposed as template fields that can be used in the bodies of emails, Slack or Teams messages.

Should be possible to add in coming Butler versions, can't give a time estimate right now though.

**Comment by mountaindude** [comment]
@wieeem 

A new version of Butler ([9.2.0](https://github.com/ptarmiganlabs/butler/releases)) has just been released. 
It adds various app owner metadata to the template fields that are available when sending notifications to Teams, Slack, email and webhooks.

A complete list of available template fields for reload task alerts is available [here](https://butler.ptarmiganlabs.com/docs/reference/alert-template-fields/reload-tasks/).
